src/main/java/org/alfresco/repo/bulkimport/metadataloaders/AbstractMapBasedMetadataLoader.java

```diff
@@ -101,14 +101,27 @@ public final String getMetadataFileExtension()
      */
     abstract protected Map<String,Serializable> loadMetadataFromFile(final Path metadataFile);
 
-
     @Override
     public final void loadMetadata(final ContentAndMetadata contentAndMetadata, Metadata metadata)
     {
         if (contentAndMetadata.metadataFileExists())
         {
             final Path metadataFile = contentAndMetadata.getMetadataFile();
+            final String metadataFilePath = FileUtils.getFileName(metadataFile);
+            try
+            {
+                loadMetadataInternal(metadata, metadataFile, metadataFilePath);
+            }
+            catch (Exception e)
+            {
+                log.error("Error encountered when reading metadata file '" + metadataFilePath + "'.");
+                throw new RuntimeException("Exception from reading file: '" + metadataFilePath + "'.", e);
+            }
+        }
+    }
 
+    private void loadMetadataInternal(Metadata metadata, Path metadataFile, String metadataFilePath)

```

Comment:
the ideal number of parameters to a method is 0. You can easily get this "metadataFilePath" information inside the method, from the other variables.

continue to add final to the remaining parameters